tests/agent/test_agent.py

```diff
@@ -290,6 +302,18 @@ def test_gx_agent_run_handles_client_probable_authentication_error_on_init(
         agent.run()
 
 
+def test_gx_agent_run_handles_amqp_error_on_init(
```

Comment:
There are similar tests above that mock the client, but not the run method itself; that setup didn't help here, so just added a new test around the new error type.